src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -291,23 +291,34 @@ private static function colorize($str, $red)
      * @param string[]      $groups
      * @param Configuration $configuration
      * @param bool          $isFailing
+     *
+     * @throws \InvalidArgumentException
      */
     private function displayDeprecations($groups, $configuration, $isFailing)
     {
         $cmp = function ($a, $b) {
             return $b->count() - $a->count();
         };
 
+        if ($configuration->shouldWriteToLogFile()) {
+            if (false === $handle = @fopen($file = $configuration->getLogFile(), 'a')) {
+                throw new \InvalidArgumentException(sprintf('The configured log file "%s" is not writeable.', $file));
+            }
+        } else {
+            $handle = fopen('php://output', 'w');
+        }
+
         foreach ($groups as $group) {
             if ($this->deprecationGroups[$group]->count()) {
-                echo "\n", self::colorize(
-                    sprintf(
-                        '%s deprecation notices (%d)',
-                        \in_array($group, ['direct', 'indirect', 'self'], true) ? "Remaining $group" : ucfirst($group),
-                        $this->deprecationGroups[$group]->count()
-                    ),
-                    'legacy' !== $group && 'indirect' !== $group
-                ), "\n";
+                $deprecationGroupMessage = sprintf(
+                    '%s deprecation notices (%d)',
+                    \in_array($group, ['direct', 'indirect', 'self'], true) ? "Remaining $group" : ucfirst($group),
+                    $this->deprecationGroups[$group]->count()
+                );
+                echo "\n", self::colorize($deprecationGroupMessage, 'legacy' !== $group && 'indirect' !== $group), "\n";
```

Comment:
it looks strange to both echo and fwrite when shouldWriteToLogFile is true
this should be in an "else" block (maybe my previous comment was confusion here)

Reply:
But do you mean for stylistic reasons? Now it is still correct and less complex than before. I could add a comment (`// If the logFile option is used we echo the colorized group message and write the uncolorized group message to the log file`) to make it clearer. If shouldWriteToLogFile then echo it colorized and also write it to the file uncolorized.

The echo is because @SimonHeimberg wrote: "I would like to be able to see a short output of deprecations" in https://github.com/symfony/symfony/issues/37679

Reply:
@nicolas-grekas Do you think it is wrong? Or is it ok?

Reply:
I think this should be updated yes: why would we echo anything when a file is configured as the output?

Reply:
Ok, thank you, the echo is removed now.